### DISCUSSION TITLE
SignInScreen blocked by keyboard

### DISCUSSION THREAD

**Discussion by fredecodes** [firebase/flutterfire#8050] [Feature Feedback]
There seems to be a really big problem with SigninScreen that I've not seen mentioned elsewhere, which surprises me as I see it as a huge usability issue.  When rendered on a regular or small sized mobile screen, the "Password" text field is completely hidden by the keyboard (on both Android and iOS).  You can still type to enter text, but you are typing in the blind because you can't see the cursor.  This is very frustrating for users, especially when it's their very first impression of your app.  When the keyboard is up, the screen should scroll to fit within the upper bounds that are visible.  Am I missing something, is there a way around this to let the user see the text field that is hidden by the keyboard?

**Comment by olibye** [comment]
This can be solved by placing the SignInScreen inside a Scaffold... https://api.flutter.dev/flutter/material/Scaffold-class.html

> The scaffold will expand to fill the available space. That usually means that it will occupy its entire window or device screen. When the device's keyboard appears the Scaffold's ancestor [MediaQuery](https://api.flutter.dev/flutter/widgets/MediaQuery-class.html) widget's [MediaQueryData.viewInsets](https://api.flutter.dev/flutter/widgets/MediaQueryData/viewInsets.html) changes and the Scaffold will be rebuilt. By default the scaffold's [body](https://api.flutter.dev/flutter/material/Scaffold/body.html) is resized to make room for the keyboard.
